v2/1.9.3/Dockerfile

```diff
@@ -0,0 +1,30 @@
+## Dockerfile for a haskell environment
+FROM       debian:stretch-20180312
+
+ENV STACK_VERSION 1.9.3
+
+## ensure locale is set during build
+ENV LANG            C.UTF-8
+
+RUN apt-get update && \
+    apt-get install -y procps curl gnupg2 ca-certificates g++ libgmp-dev libncurses-dev make xz-utils git zlib1g-dev && \
+# fetch stack and the gpg sig
+    curl -fSL https://github.com/commercialhaskell/stack/releases/download/v$STACK_VERSION/stack-$STACK_VERSION-linux-x86_64-static.tar.gz -o stack.tar.gz && \
+    curl -fSL https://github.com/commercialhaskell/stack/releases/download/v$STACK_VERSION/stack-$STACK_VERSION-linux-x86_64-static.tar.gz.asc -o stack.tar.gz.asc && \
+# setup gpg and get keys
+    export GNUPGHOME="$(mktemp -d)" && \
+    gpg --keyserver ipv4.pool.sks-keyservers.net --recv-keys C5705533DA4F78D8664B5DC0575159689BEFB442 && \
```

Comment:
This doesn't build for me at the moment, and I believe it's because `--batch` needs to be added to this line to work around upstream `gpg` breakage. I believe we are feeling the impact of:

https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=913614

Which is referenced from here:

https://github.com/nodejs/docker-node/issues/922

where they are getting the same `gpg: cannot open '/dev/tty': No such device or address` error that I am seeing.